ext/drreg/drreg.c

```diff
@@ -65,17 +65,49 @@
 
 #define PRE instrlist_meta_preinsert
 
-/* This should be pretty hard to exceed as there aren't this many GPRs */
+/* We add additional spill slots to cater for client spills.
```

Comment:
Both SPILL_SLOT_MAX (DR slots?) and 8 could be explained further. I think it is because SPILL_SLOT_MAX are the available client spill slots and we're adding yet another 8 for another set of gprs?

Reply:
The extra slots are used to as temporary scratch in order to store client values when drreg is handling cross-instruction reservation.

Reply:
> The extra slots are used as temporary scratch in order to store client values when drreg is handling cross-instruction reservation and restoration of the original value is in required.

No. The client needs to reserve extra slots for this. The number here is just the hardcoded maximum number of spills we support and can track.

I am updating the PR here right now with your permission (just in case so we're not overlapping work)

Reply:
Sure, I am not modifying the PR. What are the extra slots (which yes the user needs to allocate) used for then? I am having another look at the code now.

Reply:
I think this is best explained if one looks at find_free_slot(per_thread_t *pt) and its usage.

Reply:
The tool has to request the extra slot. Maybe the 8 *was* added to always have space for this. Maybe that's what you meant.

Reply:
So the subject is how are these slots set-up and not what they are used for? In this case, yes, the tool has to request extra addressable TLS storage. However, if there aren't enough addressable TLS slots to accommodate SPILL_SLOT_MAX slots, drreg defaults to using dr_slots for remaining slots. Look at spill_ and restore_directly functions to see this functionality.

Reply:
I think I resolved this issue?

Reply:
Resolving this issue, unresolve if needed.